.github/workflows/health_checks.yml

```diff
@@ -298,9 +317,10 @@ jobs:
   update_or_publish_versions:
     if: ${{ github.event_name == 'push' && github.ref_name == 'main' }}
     needs:
-      - run_package_manager_e2e_tests
       - test_with_coverage
-      - run_e2e_tests
+      - e2e_package_manager
+      - e2e_deployment
+      - e2e_create_amplify
```

Comment:
I think it's a good change to distribute failure risk into multiple jobs. This means a probablity of getting green build in 1-3 retries stays flat while we add tests.

However. It sounds like we should think about `split-e2e-tests` + PR check for that (like we have in gen1). If not in this PR we should log GH issue to follow up.

Reply:
I'd like to go as far as possible using the matrix feature of GH actions before we start generating the workflow files. Reason being I think it makes it much more cumbersome to figure out how / where to add config to the workflow. I'll cut an issue to think more about this as the need arises.

Reply:
at this point. I'm mostly interested in something that can assert that all tests are included in the workflow.
Which might be less headache than full blown generation.

Reply:
https://github.com/aws-amplify/amplify-backend/issues/1190